### PR TITLE
Feature/inventory envs

### PR DESCRIPTION
**Pull request by trasher**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes

### CODE REVIEW

File: src/Item_Environment.php [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+            'is_tab' => true,
+            'items_id' => $items_id,
+            'filters' => Sanitizer::dbEscapeRecursive($filters),
+            'columns' => [
+                'key'           => __("Key"),
+                'value'           => __("Value"),
+            ],
+            'columns_values' => [
+                'user' => $users,
+            ],
```
Comment (orthagh) [inline src/Item_Environment.php:158]:
```suggestion
```
You don't have any `user` entry in `columns` key

File: src/Item_Environment.php [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+            }
+        }
+
+        return $sql_filters;
+    }
+
+
+    public static function getIcon()
+    {
+        return "ti ti-bolt";
```
Comment (orthagh) [inline src/Item_Environment.php:186]:
```suggestion
        return "ti ti-terminal-2";
```

**Review by orthagh — CHANGES_REQUESTED** [review]
Maybe also `Phone` could have the new tab

File: src/Computer.php [hunk 3]
```diff
@@ -109,6 +110,7 @@ public function defineTabs($options = [])
          ->addStandardTab('Item_Disk', $ong, $options)
          ->addStandardTab('Item_SoftwareVersion', $ong, $options)
          ->addStandardTab('Item_Process', $ong, $options)
+        ->addStandardTab('Item_Environment', $ong, $options)
```
Comment (orthagh) [inline src/Computer.php:113]:
```suggestion
         ->addStandardTab('Item_Environment', $ong, $options)
```

**Comment by trasher** [comment]
> Maybe also `Phone` could have the new tab

Tab is already present
image

**Comment by orthagh** [comment]
Okay, i did the comment a little too quick, in my mind it was also to add the clone relation and the purge of sub envs if needed

**Comment by trasher** [comment]
> Okay, i did the comment a little too quick, in my mind it was also to add the clone relation and the purge of sub envs if needed

It's also done. There are 3 `Item_Environment` ocurrences in Computer.php and also 3 in Phone.php

**Review by orthagh — APPROVED** [review]

File: src/Inventory/Asset/Environment.php [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+ * ---------------------------------------------------------------------
+ */
+
+namespace Glpi\Inventory\Asset;
+
+use Glpi\Inventory\Conf;
+use Item_Environment;
+use Toolbox;
+
+class Environment extends InventoryAsset
```
Comment (cedric-anne) [inline src/Inventory/Asset/Environment.php:43]:
```suggestion
final class Environment extends InventoryAsset
```

File: src/Inventory/Asset/Environment.php [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+                    $itemEnv->update(Toolbox::addslashes_deep($input));
+                    unset($value[$key]);
+                    unset($db_itemEnvs[$keydb]);
+                    break;
+                }
+            }
+        }
+
+        if ((!$this->main_asset || !$this->main_asset->isPartial()) && count($db_itemEnvs) != 0) {
+           // Delete Item_Process in DB
```
Comment (cedric-anne) [inline src/Inventory/Asset/Environment.php:111]:
```suggestion
           // Delete Item_Environment in DB
```

File: src/Item_Environment.php [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+ * ---------------------------------------------------------------------
+ */
+
+use Glpi\Application\View\TemplateRenderer;
+use Glpi\Toolbox\Sanitizer;
+
+/**
+ * Environment Class
+ **/
+class Item_Environment extends CommonDBChild
```
Comment (cedric-anne) [inline src/Item_Environment.php:42]:
```suggestion
final class Item_Environment extends CommonDBChild
```

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**